packages/sit-onyx/src/components/OnyxDataGrid/features/renderer.ts

```diff
@@ -7,13 +7,45 @@ import "./renderer.scss";
 
 export type DefaultSupportedTypes = "string" | "number";
 
-const numberFormatter = <TEntry extends DataGridEntry>(value: TEntry[keyof TEntry] | undefined) => {
+export const FALLBACK_RENDER_VALUE = "-" as const;
+
+const numberFormatter = <TEntry extends DataGridEntry>(
+  value: TEntry[keyof TEntry] | undefined,
+): string => {
+  // using loose "==" here to catch both undefined and null
+  if (
+    value == undefined ||
+    value instanceof Date ||
+    typeof value === "boolean" ||
+    typeof value === "symbol"
+  ) {
+    return FALLBACK_RENDER_VALUE;
+  }
+
   const locale = injectI18n().locale;
   const formatter = new Intl.NumberFormat(locale.value);
 
   // We format the given value as Number. In case it renders as NaN, we replace it with `-`.
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- The typing is incorrect, the `format` method accepts any value
-  return formatter.format(value as any).replace("NaN", "-");
+  // The typing is incorrect, the `format` method accepts any value
+  return formatter
+    .format(value as Parameters<typeof formatter.format>[0])
```

Comment:
If we want to use a complex type like this, then I would prefer to extract it into a explicit, named type.
Also we can expect the type of the Intl API to be very stable, so IMHO we can also just do:

```suggestion
    .format(value as number)
```
or

```suggestion
    .format(value as (number | bigint))
```